File: day02/day02.py
input = open('input.txt', 'r').read().split('\n')
ex = open('example.txt', 'r').read().split('\n')


def solve(data):
    xpos1 = 0
    depth1 = 0
    for i in range(0, len(data) - 1):
        if data[i][0] == 'f':
            xpos1 += int(data[i][-1])
        elif data[i][0] == 'd':
            depth1 += int(data[i][-1])
        elif data[i][0] == 'u':
            depth1 -= int(data[i][-1])

    xpos2 = 0
    depth2 = 0
    aim = 0
    for i in range(0, len(data)-1):
        if data[i][0] == 'f':
            xpos2 += int(data[i][-1])
            depth2 += (aim * int(data[i][-1]))
        elif data[i][0] == 'd':
            aim += int(data[i][-1])
        elif data[i][0] == 'u':
            aim -= int(data[i][-1])

    return xpos1 * depth1, xpos2 * depth2

File: day02/day02.m
    data = readtable('input.txt', 'Delimiter', 'space');
    dir = string(table2array(data(:,1)));
    mag = table2array(data(:,2));
    
    disp("Part 1: " + part1(dir, mag));
    disp("Part 2: " + part2(dir, mag));

    % Function to solve p1
    function answer = part1(dir, mag)
        xpos = 0;
        depth = 0;

        for i = 1:length(dir)
            if dir(i) == "forward"
                xpos = xpos + mag(i);
            elseif dir(i) == "up"
                depth = depth - mag(i);
            elseif dir(i) == "down"
                depth = depth + mag(i);
            end
        end
        answer = xpos * depth;
    end

    % Function to solve p2
    function answer = part2(dir, mag)
        xpos = 0;
        depth = 0;
        aim = 0;    
        
        for i = 1:length(dir)
            if dir(i) == "forward"
                xpos = xpos + mag(i);
                depth = depth + aim * mag(i);
            elseif dir(i) == "up"
                aim = aim - mag(i);
            elseif dir(i) == "down"
                aim = aim + mag(i);
            end
        end
        answer = xpos * depth;
    end
